File: backend/app/api/assets.py
import uuid

from fastapi import APIRouter

from app.schemas.assets import GenerateRequest, GenerateResponse
from app.services.local_generator import local_generator
from app.services.prompt_builder import build_prompt

router = APIRouter()


@router.post("/generate", response_model=GenerateResponse)
def generate_assets(request: GenerateRequest) -> GenerateResponse:
    enhanced_prompt, constraints = build_prompt(request)
    assets = local_generator.generate(request, enhanced_prompt)
    return GenerateResponse(
        request_id=str(uuid.uuid4()),
        enhanced_prompt=enhanced_prompt,
        constraints=constraints,
        assets=assets,
    )

File: frontend/src/api/assets.ts
import axios from 'axios'
import type { GenerateRequest, GenerateResponse } from '../types/assets'

const apiClient = axios.create({
  baseURL: import.meta.env.VITE_API_BASE_URL ?? '/api',
  timeout: 30000,
})

const toApiPayload = (payload: GenerateRequest) => ({
  project_name: payload.projectName,
  description: payload.description,
  asset_type: payload.assetType,
  style: payload.style,
  size: payload.size,
  count: payload.count,
  transparent_background: payload.transparentBackground,
  palette: payload.palette,
  consistency_seed: payload.consistencySeed,
})

const toClientAsset = (asset: any) => ({
  id: asset.id,
  name: asset.name,
  assetType: asset.asset_type,
  style: asset.style,
  size: asset.size,
  dataUrl: asset.data_url,
  seed: asset.seed,
  prompt: asset.prompt,
  createdAt: asset.created_at,
  metadata: asset.metadata,
})

export const generateAssets = async (payload: GenerateRequest): Promise<GenerateResponse> => {
  const { data } = await apiClient.post('/assets/generate', toApiPayload(payload))

  return {
    requestId: data.request_id,
    enhancedPrompt: data.enhanced_prompt,
    constraints: data.constraints,
    assets: data.assets.map(toClientAsset),
  }
}

export const checkBackendHealth = async () => {
  const { data } = await apiClient.get('/health')
  return data
}
